// pages/createArticle/index.js
import React, { useEffect, useState, useContext } from 'react'
import { Redirect } from 'react-router-dom'

import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import { CurrentUserContext } from 'context/currentUser'

const CreateArticle = () => {
	const apiUrl = '/articles'
	const [{ response, error }, doFetch] = useFetch(apiUrl)
	const [currentUserContext] = useContext(CurrentUserContext)
  const inititalValues = {
		title: '',
		description: '',
		body: '',
		tagList: []
	}
	const [isSuccessFullSubmit, setIsSuccessFullSubmit] = useState(false)
  const handleSubmit = article => {
		doFetch({
			method: 'post',
			data: {
				article
			}
		})
	}
	
	useEffect(()=> {
		if (response) {
			setIsSuccessFullSubmit(true)
		}
	}, [response])

	if (currentUserContext.isLoggedIn === false) {
		return <Redirect to='/' />
	}


	if (isSuccessFullSubmit) {
		return <Redirect to={`/articles/${response.article.slug}`} />
	}

  return (
    <div>
			<ArticleForm 
				errors={(error && error.errors) || {}} 
				inititalValues={inititalValues} 
				onSubmit={handleSubmit} 
			/>
    </div>
  )
}

export default CreateArticle
