// pages/editArticle/index.js
import React, {useContext, useEffect, useState} from 'react'
import { Redirect } from 'react-router-dom'

import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import { CurrentUserContext } from 'context/currentUser'

const EditArticle = ({match}) => {

  const slug = match.params.slug
  const apiUrl = `/articles/${slug}`
  const [currentUserContext] = useContext(CurrentUserContext)
  const [{ response: fetchArticleResponse }, doFetchArticle] = useFetch(apiUrl)
  const [
    {
      response: updateArticleResponse,
      error: updateArticleError
    },
    doUpdateArticle
  ] = useFetch(apiUrl)

  const [initialValues, setInitialValues] = useState(null)
  const [isSuccessFullSubmit, setIsSuccessFullSubmit] = useState(false)

  const handleSubmit = article => {
    doUpdateArticle({
      method: 'put',
      data: {
        article
      }
    })
  }

  useEffect(() => {
    doFetchArticle()
  }, [doFetchArticle])

  useEffect(() => {
    if (!fetchArticleResponse) {
      return
    }

    setInitialValues({
      title: fetchArticleResponse.article.title,
      description: fetchArticleResponse.article.description,
      body: fetchArticleResponse.article.body,
      tagList: fetchArticleResponse.article.tagList
    })

  }, [fetchArticleResponse])

  useEffect(() => {
    if (!updateArticleResponse) {
      return
    }
    setIsSuccessFullSubmit(true)
  }, [updateArticleResponse])

  if (currentUserContext.isLoggedIn === false) {
    return <Redirect to='/' />
  }


  if (isSuccessFullSubmit) {
    return <Redirect to={`/articles/${slug}`} />
  }

  return (
    <ArticleForm
      onSubmit={handleSubmit}
      errors={(updateArticleError && updateArticleError.errors) || {}}
      initialValues={initialValues}
    />
  )
}

export default EditArticle
